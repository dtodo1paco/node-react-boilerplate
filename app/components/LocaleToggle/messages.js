/*
 * LocaleToggle Messages
 *
 * This contains all the text for the LanguageToggle component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  label: {
    id: 'app.containers.LocaleToggle.label',
    defaultMessage: 'Language',
  },
  help: {
    id: 'app.containers.LocaleToggle.help',
    defaultMessage: 'Need another language, please help us translating',
  },
  en: {
    id: 'app.containers.LocaleToggle.en',
    defaultMessage: 'en',
  },
  es: {
    id: 'app.containers.LocaleToggle.es',
    defaultMessage: 'es',
  },
});
