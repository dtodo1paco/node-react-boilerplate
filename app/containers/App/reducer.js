import { fromJS } from 'immutable';
import { saveUser, getUser } from 'api/storage';
import {
  SET_LOADING,
  SET_NOTIFICATION,
  AUTH_USER_START,
  AUTH_REAUTH_USER_START,
  AUTH_USER_OK,
  AUTH_LOGOUT_START,
  SESSION_EXPIRED,
} from './constants';

// for testing purposes
export const loggedUser = {
  auth: true,
  token:
    'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjViYTExZDExYjlmNWQ4NWI3OTYxM2Y2YyIsImlhdCI6MTUzOTkyNzYzNSwiZXhwIjoxNTM5OTI3OTM1fQ.bt4jv-tecy2qjRgC9S5xTkl1PkY7bN_BwaWSWIVbu_Q',
  data: {
    userid: '44',
    name: '44',
    email: '11@ascasd.co',
    lastLogin: '2018-10-18T16:12:03.565Z',
    settings: {
      theme: 'material',
      locale: 'es',
      lastRoute: '/account/settings',
    },
  },
};

export const anonymousUser = {
  auth: false,
  token: null,
  data: null,
};
const currentUser = {
  auth: false,
  token: getUser(),
  data: null,
};
export const initialState = fromJS({
  user: currentUser,
  notification: null,
  loading: false,
});

function appReducer(state = initialState, action) {
  switch (action.type) {
    case AUTH_LOGOUT_START:
    case AUTH_REAUTH_USER_START:
    case AUTH_USER_START:
      return state.merge({
        loading: true,
        user: anonymousUser,
      });
    case SESSION_EXPIRED:
      saveUser(null);
      return state.merge({
        loading: false,
        user: anonymousUser,
        notification: action.payload,
      });
    case AUTH_USER_OK:
      saveUser(action.payload);
      return state.merge({
        loading: false,
        user: action.payload,
      });
    case SET_NOTIFICATION:
      return state.merge({
        loading: false,
        notification: action.payload,
      });
    case SET_LOADING:
      return state.set('loading', action.payload);
    default:
      return state;
  }
}
export default appReducer;
