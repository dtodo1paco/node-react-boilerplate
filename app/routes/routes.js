import React from 'react';

import HomeApp from 'containers/HomeApp/Loadable';
import HomePage from 'containers/HomePage/Loadable';
import AuthPage from 'containers/AuthPage/index';
import Logout from 'components/Logout';
import UserSettings from 'containers/UserSettings';

import AccountCircle from '@material-ui/icons/AccountCircle';
import HelpIcon from '@material-ui/icons/Help';
import HomeIcon from '@material-ui/icons/Home';
import ExitToApp from '@material-ui/icons/ExitToApp';

// TODOS del sistema de navegación
// TODO C: anidar menús
//
export const PUBLIC_NAVIGATION_ITEMS = [
  {
    id: 'home',
    icon: <HomeIcon />,
    to: null,
    path: '/',
    component: HomePage,
  },
  {
    id: 'login',
    icon: <AccountCircle />,
    to: '/auth/login',
    path: '/auth/login',
    component: AuthPage,
  },
  {
    id: 'register',
    icon: null,
    to: null, // we don't want a link to register page (LoginPage will include it)
    path: '/auth/register',
    component: AuthPage,
  },
  {
    id: 'help',
    icon: <HelpIcon />,
    to: '/help',
    path: '/help',
    component: null,
  },
  /*
  Features,
  Benefits,
  Pricing,
  ...
   */
];

export const USER_ACCOUNT_ITEMS = [
  {
    id: 'app',
    icon: <HomeIcon />,
    to: '/app',
    path: '/app',
    component: HomeApp,
  },
  {
    id: 'settings',
    icon: <AccountCircle />,
    to: '/account/settings',
    path: '/account/settings',
    component: UserSettings,
  },
  {
    id: 'logout',
    icon: <ExitToApp />,
    to: '/auth/logout',
    path: '/auth/logout',
    component: Logout,
  },
];

export const PRIVATE_NAVIGATION_ITEMS = [...USER_ACCOUNT_ITEMS]; // <--- include here more groups of routes

export function getRoutesForRole(user) {
  if (!user.auth) {
    return PUBLIC_NAVIGATION_ITEMS;
  } // TODO: return routes group based on user info (role, permissions,...whatever...)
  return PRIVATE_NAVIGATION_ITEMS;
}
/**
 * True if the specified path is included in routes array
 * @param routes - array of routes (see above definitions)
 * @param path - string to search in the array of routes
 * @returns {boolean}
 */
export const routeExists = (routes, path) =>
  filterRoutes(routes, path).length > 0;

export const filterRoutes = (routes, path) =>
  routes.filter(el => el.path === path);
