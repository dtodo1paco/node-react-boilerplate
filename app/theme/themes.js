import { createTheme } from './util';
/* COLORS AVAILABLE */
import DEFAULT_COLORS from './default_theme';
import MATERIAL_COLORS from './material';

export const DEFAULT_THEME = createTheme('default', 'dark', DEFAULT_COLORS);
export const MATERIAL_THEME = createTheme('material', 'light', MATERIAL_COLORS);
export const THEMES = {
  default: DEFAULT_THEME,
  material: MATERIAL_THEME,
};

export const findThemeOrDefault = themeId =>
  THEMES[themeId] ? THEMES[themeId] : DEFAULT_THEME;
