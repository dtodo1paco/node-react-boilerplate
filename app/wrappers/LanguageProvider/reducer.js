/*
 *
 * LanguageProvider reducer
 *
 */

import { fromJS } from 'immutable';

import { CHANGE_LOCALE } from './constants';
import { DEFAULT_LOCALE } from '../../i18n'; // eslint-disable-line

export const initialState = fromJS({
  locale: DEFAULT_LOCALE,
});

function languageProviderReducer(state = initialState, action) {
  switch (action.type) {
    case CHANGE_LOCALE:
      if (action.payload === state.get('locale')) {
        return state;
      }
      return state.set('locale', action.payload);
    default:
      return state;
  }
}

export default languageProviderReducer;
