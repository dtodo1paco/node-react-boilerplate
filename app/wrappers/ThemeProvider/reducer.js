/*
 *
 * ThemeProvider reducer
 *
 */
import { fromJS } from 'immutable';
import { DEFAULT_THEME } from 'theme/themes';
import { CHANGE_THEME } from './constants';

export const initialState = fromJS(DEFAULT_THEME);

function themeProviderReducer(state = initialState, action) {
  switch (action.type) {
    case CHANGE_THEME:
      if (action.payload.props.id === state.get('props').get('id')) {
        return state;
      }
      return state.merge(action.payload);
    default:
      return state;
  }
}

export default themeProviderReducer;
