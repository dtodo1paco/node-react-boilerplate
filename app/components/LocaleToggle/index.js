/*
 *
 * LanguageToggle
 *
 */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';
import { withStyles } from '@material-ui/core/styles';
import { FormattedMessage } from 'react-intl';
import { isMobile } from 'react-device-detect';

import { appLocales } from 'i18n';
import { changeLocale } from 'wrappers/LanguageProvider/actions';
import { makeSelectLocale } from 'wrappers/LanguageProvider/selectors';

import InputLabel from '@material-ui/core/InputLabel';
import FormHelperText from '@material-ui/core/FormHelperText';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';

import messages from './messages';
const styles = theme => ({
  inAppBar: {
    margin: theme.spacing.unit,
    '& *': {
      fontSize: 'x-small',
    },
  },
});

export class LocaleToggle extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      options: this.setOptions(),
    };
  }

  componentWillReceiveProps(nextProps) {
    if (this.props.locale !== nextProps.locale) {
      this.setOptions();
    }
  }

  setOptions() {
    const opts = appLocales.map(key => (
      <option value={key} key={key}>
        <FormattedMessage {...messages[key]} />
      </option>
    ));
    return opts;
  }

  handleChange = event => {
    this.props.onLocaleToggle(event.target.value);
  };

  render() {
    const { inAppBar, classes, locale } = this.props;
    const { options } = this.state;
    const select = (
      <Select
        native={isMobile}
        value={locale}
        onChange={this.handleChange}
        name="locale"
        inputProps={{
          id: 'locale',
        }}
      >
        {options}
      </Select>
    );
    if (inAppBar) {
      return <div className={classes.inAppBar}>{select}</div>;
    }
    return (
      <div>
        <FormControl required>
          <InputLabel htmlFor="locale">
            <FormattedMessage {...messages.label} />
          </InputLabel>
          {select}
          <FormHelperText>
            <FormattedMessage {...messages.help} />
          </FormHelperText>
        </FormControl>
      </div>
    );
  }
}

LocaleToggle.propTypes = {
  onLocaleToggle: PropTypes.func,
  locale: PropTypes.string.isRequired,
  theme: PropTypes.object.isRequired,
  classes: PropTypes.object.isRequired,
  inAppBar: PropTypes.bool,
};

const mapStateToProps = createStructuredSelector({
  locale: makeSelectLocale(),
});
export const mapDispatchToProps = dispatch => ({
  onLocaleToggle: locale => dispatch(changeLocale(locale)),
  dispatch,
});
const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);
const themeConnected = withStyles(styles, { withTheme: true });

export default compose(
  themeConnected,
  withConnect,
)(LocaleToggle);
