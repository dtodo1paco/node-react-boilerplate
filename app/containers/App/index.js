/**
 *
 * App.js
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 *
 * NOTE: while this component should technically be a stateless functional
 * component (SFC), hot reloading does not currently support SFCs. If hot
 * reloading is not a necessity for you then you can refactor it and remove
 * the linting exception.
 */
/* eslint-disable import/first */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';
import injectReducer from 'utils/injectReducer';
import { withStyles } from '@material-ui/core/styles';
import classnames from 'classnames';
import { isNil } from 'lodash';

import reducer from './reducer';
import saga from './saga';
import injectSaga from 'utils/injectSaga';
import {
  makeSelectLoading,
  makeSelectNotification,
  makeSelectUser,
  makeSelectLocation,
} from './selectors';
import { authReauthUserStart, setAppNotification } from './actions';

import { makeSelectLocale } from 'wrappers/LanguageProvider/selectors';

import Header from 'components/Header';
import Footer from 'components/Footer';
import { getRoutesForRole } from 'routes/routes';
import { Switch, Route } from 'react-router-dom';
import CircularProgress from '@material-ui/core/CircularProgress';
import Notification from 'components/Notification';
import NotFoundPage from 'containers/NotFoundPage/Loadable';

import diff from 'deep-diff'; // TODO: remove when console.log disappear
/**
 * Maps navItems for user-role to Routes to navigate (if navItems has a path)
 * @param navItems
 * @returns Route for every navItems with path defined
 */
const mapNavItems = function cb(navItems) {
  return navItems.map(
    item =>
      item.path ? (
        <Route exact path={item.path} component={item.component} />
      ) : null,
  );
};

const styles = theme => ({
  content: {
    marginLeft: theme.spacing.unit,
    marginRight: theme.spacing.unit,
  },
  root: {
    backgroundColor: theme.palette.background.default,
  },
});

export class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      content: null,
    };
  }

  isUserToken(user) {
    return user.auth === false && user.token !== null;
  }

  componentWillMount() {
    // eslint-disable-next-line
    console.log('!!!!!! APP WILL MOUNT !!!!!!');
    if (this.isUserToken(this.props.user)) {
      // eslint-disable-next-line
      console.log('Retrieving user from server...');
      this.props.authReauthUserStart();
    }
  }

  componentDidMount() {
    if (!this.isUserToken(this.props.user)) {
      this.setContentForUser(this.props.user);
    }
    /*
    if (this.isUserToken(this.props.user)) {
      console.log("componentDidMount: We are waiting for user to come... then content will be set properly");
    } else { // no previous login
      console.log("componentDidMount: no previous login");
      this.setContentForUser(this.props.user);
    }
    */
  }

  componentWillReceiveProps(nextProps) {
    const hasLoggedIn = !this.props.user.auth && nextProps.user.auth;
    const hasLoggedOut = this.props.user.auth && !nextProps.user.auth;
    const loginFailed =
      !this.props.user.auth &&
      !nextProps.user.auth &&
      nextProps.notification != null;
    const sessionExpired = hasLoggedOut && nextProps.notification != null;

    const userLoggedX = hasLoggedIn || hasLoggedOut;
    // console.log(">> props arrives");
    // this.printProps(nextProps);
    // console.log("loggedIn ["+hasLoggedIn+"] loginFailed ["+loginFailed+"] loggedOut ["+hasLoggedOut+"] sessionExpired ["+sessionExpired+"]");

    let route = nextProps.location.pathname;
    if (!nextProps.loading && !this.state.content) {
      // console.log("props says --- we need some content");
      this.setContentForUser(nextProps.user);
      if (loginFailed) {
        route = '/auth/login';
      }
    } else if (userLoggedX) {
      // console.log("props says --- you logged X -- content is: " + this.state.content);
      this.setContentForUser(nextProps.user);
      /* eslint-disable */
      route = nextProps.user.auth ?
                nextProps.user.data.settings ?
                  nextProps.user.data.settings.lastRoute :
                  '/app' :
              loginFailed ?
                '/auth/login' :
                sessionExpired ?
                  '/auth/login' :
                  '/';
      /* eslint-enable */
    }
    if (route !== nextProps.location.pathname) {
      // eslint-disable-next-line
      console.log(`====> navigating to: ${route}`);
      this.context.router.history.push(route);
    }
  }

  setContentForUser(user) {
    // eslint-disable-next-line
    const routes = getRoutesForRole(user);
    // console.log("  -* SET CONTENT *- [auth:"+JSON.stringify(user.auth)+"] content ["+routes.length+"]");
    this.setState({
      content: mapNavItems(routes),
    });
  }

  shouldComponentUpdate(nextProps, nextState) {
    /* eslint-disable */
    const contentChanged = !isNil(nextState.content) && this.state.content !== nextState.content;
    const userLoggedX = !this.isUserToken(nextProps.user) && !this.isUserToken(this.props.user) && this.props.user != nextProps.user;
    const loadingChanged = this.props.loading !== nextProps.loading;
    const locationChanged = this.props.location.pathname !== nextProps.location.pathname;
    const themeChanged = this.props.theme.props.id != nextProps.theme.props.id;
    const classesChanged = this.props.classes.content != nextProps.classes.content;
    const localeChanged = this.props.locale != nextProps.locale;
    const scu = locationChanged || loadingChanged || userLoggedX || contentChanged || classesChanged || themeChanged || localeChanged;
    if (!scu) { // TODO: remove also import of deep-diff
      console.log("  scu: ···skip rendering app. WHY ?");
      console.log(JSON.stringify(diff(this.props, nextProps)));
      console.log("  -----");
      console.log(JSON.stringify(diff(this.state, nextState)));
    }
    /* eslint-enable */
    return scu;
  }

  render() {
    const { content } = this.state;
    const { notification, loading, location, classes } = this.props;
    /*
    const isNull = !content ? " --- NULL ---": '';
    console.log("*** RENDER ["+this.props.location.pathname+"] "+isNull+"  user Auth: " + user.auth + " loading ["+loading+"] notif ["+notification+"] theme ["+theme.props.id+"] locale ["+locale+"]");
    */
    if (!content) return null;
    return (
      <div className={classnames('appRoot', classes.root)}>
        <Header />
        <main id="content" className={classes.content}>
          {loading && (
            <div className="progress">
              <CircularProgress color="secondary" />
            </div>
          )}
          {!loading && (
            <Switch location={location}>
              {content}
              <Route component={NotFoundPage} />
            </Switch>
          )}
          {notification && (
            <Notification
              open
              variant={notification.type}
              data={notification}
              onClose={this.props.resetNotification}
            />
          )}
        </main>
        <Footer />
      </div>
    );
  }
}

// get access to router and history to push routes if necessary
App.contextTypes = { router: PropTypes.object };

App.propTypes = {
  authReauthUserStart: PropTypes.func.isRequired,
  resetNotification: PropTypes.func.isRequired,
  notification: PropTypes.object.isRequired,
  loading: PropTypes.bool.isRequired,
  // eslint-disable-next-line
  location: PropTypes.object.isRequired,
  user: PropTypes.object.isRequired,
  classes: PropTypes.object.isRequired,
  theme: PropTypes.object.isRequired,
  locale: PropTypes.string,
};

const mapStateToProps = createStructuredSelector({
  notification: makeSelectNotification(),
  loading: makeSelectLoading(),
  location: makeSelectLocation(),
  user: makeSelectUser(),
  locale: makeSelectLocale(),
});
const mapDispatchToProps = dispatch => ({
  authReauthUserStart: data => dispatch(authReauthUserStart(data)),
  resetNotification: () => dispatch(setAppNotification(null)),
});

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);
const withReducer = injectReducer({ key: 'app', reducer });
const withSaga = injectSaga({ key: 'app', saga });
const themeConnected = withStyles(styles, { withTheme: true });

export default compose(
  withReducer,
  withSaga,
  themeConnected,
  withConnect,
)(App);
