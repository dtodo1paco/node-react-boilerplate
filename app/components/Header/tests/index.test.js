import React from 'react';
import TestHelper from 'utils/testHelper';
import { DEFAULT_THEME } from 'theme/themes';
import { anonymousUser, loggedUser } from 'containers/App/reducer';
import LocaleToggle from 'components/LocaleToggle';
import { getRoutesForRole, routeExists } from 'routes/routes';
import { Header } from '../index';

describe('<Header />', () => {
  describe('when no user logged in...', () => {
    const noLoginProps = {
      user: anonymousUser,
      location: {
        hash: 'tEsT',
        pathname: '/',
        search: '',
      },
      theme: DEFAULT_THEME,
      classes: {},
      locale: TestHelper.DEFAULT_LOCALE,
    };
    const publicHeader = <Header {...noLoginProps} />;
    const anonymousNavItems = getRoutesForRole(anonymousUser);
    const anonymousLinks = anonymousNavItems.filter(el => el.to !== null);

    it('snapshot', () => {
      const wrapper = TestHelper.shallow(publicHeader);
      expect(wrapper).toMatchSnapshot();
    });
    it('should render a div as base element', () => {
      const renderedComponent = TestHelper.shallow(publicHeader);
      expect(renderedComponent.type()).toEqual('div');
    });
    it('should render title', () => {
      const renderedComponent = TestHelper.shallow(publicHeader);
      expect(renderedComponent.find('#header-title').length).toBe(1);
    });
    it('should render nav bar', () => {
      const renderedComponent = TestHelper.shallow(publicHeader);
      expect(renderedComponent.find('#header-links').length).toBe(1);
    });
    it(`should render nav bar with ${anonymousLinks.length} links and correct "to" props`, () => { // eslint-disable-line
      const renderedComponent = TestHelper.shallow(publicHeader);
      const headerLinks = renderedComponent
        .find('#header-links')
        .children()
        .first()
        .children();
      expect(headerLinks).toHaveLength(anonymousLinks.length);
      headerLinks.forEach(node => {
        const linkRoute = node.props('key').to;
        expect(routeExists(anonymousNavItems, linkRoute)).toBeTruthy();
      });
    });
    it('should render 1 LocaleToggle', () => {
      const renderedComponent = TestHelper.shallow(publicHeader);
      expect(renderedComponent.find(LocaleToggle)).toHaveLength(1);
    });
  });
  describe('when a user is logged in...', () => {
    const loggedInProps = {
      user: loggedUser,
      location: {
        hash: 'tEsT',
        pathname: '/app',
        search: '',
      },
      theme: DEFAULT_THEME,
      classes: {},
      locale: TestHelper.DEFAULT_LOCALE,
    };
    const privateHeader = <Header {...loggedInProps} />;
    const privateNavItems = getRoutesForRole(loggedUser);
    const privateLinks = privateNavItems.filter(el => el.to !== null);

    it('snapshot', () => {
      const wrapper = TestHelper.shallow(privateHeader);
      expect(wrapper).toMatchSnapshot();
    });
    it('should render title', () => {
      const renderedComponent = TestHelper.shallow(privateHeader);
      expect(renderedComponent.find('#header-title').length).toBe(1);
    });
    it(`should render nav bar with ${privateLinks.length} links and correct "to" props`, () => { // eslint-disable-line
      const renderedComponent = TestHelper.shallow(privateHeader);
      const headerLinks = renderedComponent
        .find('#header-drawer-links')
        .children('[activeClassName="active"]');
      expect(headerLinks).toHaveLength(privateLinks.length);
      headerLinks.forEach(node => {
        const linkRoute = node.props('key').to;
        expect(routeExists(privateNavItems, linkRoute)).toBeTruthy();
      });
    });
    it('should NOT render a LocaleToggle', () => {
      const renderedComponent = TestHelper.shallow(privateHeader);
      expect(renderedComponent.find(LocaleToggle)).toHaveLength(0);
    });
  });
});
