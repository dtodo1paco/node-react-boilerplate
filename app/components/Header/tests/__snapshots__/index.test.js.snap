// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> when a user is logged in... snapshot 1`] = `
<div>
  <WithStyles(AppBar)
    className=""
    position="absolute"
  >
    <WithStyles(Toolbar)
      disableGutters={true}
    >
      <WithStyles(IconButton)
        aria-label="Open drawer"
        color="inherit"
        onClick={[Function]}
      >
        <pure(MenuIcon) />
      </WithStyles(IconButton)>
      <nav>
        <div
          id="header-title"
        >
          <Link
            replace={false}
            to="/"
          >
            <WithStyles(Typography)
              variant="headline"
            >
              <FormattedMessage
                defaultMessage="dtodo1paco"
                id="app.title"
                values={Object {}}
              />
            </WithStyles(Typography)>
          </Link>
        </div>
        <div
          id="header-links"
        >
          <div />
          <WithStyles(UserMenu)
            user={
              Object {
                "auth": true,
                "data": Object {
                  "email": "11@ascasd.co",
                  "lastLogin": "2018-10-18T16:12:03.565Z",
                  "name": "44",
                  "settings": Object {
                    "lastRoute": "/account/settings",
                    "locale": "es",
                    "theme": "material",
                  },
                  "userid": "44",
                },
                "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjViYTExZDExYjlmNWQ4NWI3OTYxM2Y2YyIsImlhdCI6MTUzOTkyNzYzNSwiZXhwIjoxNTM5OTI3OTM1fQ.bt4jv-tecy2qjRgC9S5xTkl1PkY7bN_BwaWSWIVbu_Q",
              }
            }
          />
        </div>
      </nav>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
  <WithStyles(Drawer)
    ModalProps={
      Object {
        "onBackdropClick": [Function],
      }
    }
    id="header-drawer-links"
    onClick={[Function]}
    open={false}
  >
    <div>
      <WithStyles(Typography)
        variant="headline"
      >
        <FormattedMessage
          defaultMessage="dtodo1paco"
          id="app.title"
          values={Object {}}
        />
      </WithStyles(Typography)>
      <WithStyles(IconButton)>
        <pure(ChevronLeftIcon) />
      </WithStyles(IconButton)>
    </div>
    <WithStyles(Divider) />
    <NavLink
      activeClassName="active"
      aria-current="page"
      key="app"
      location={
        Object {
          "hash": "tEsT",
          "pathname": "/app",
          "search": "",
        }
      }
      to="/app"
    >
      <WithStyles(ListItem)>
        <WithStyles(ListItemIcon)>
          <pure(HomeIcon) />
        </WithStyles(ListItemIcon)>
        <WithStyles(ListItemText)
          disableTypography={true}
        >
          <FormattedMessage
            defaultMessage="Home"
            id="app.routes.app"
            values={Object {}}
          />
        </WithStyles(ListItemText)>
      </WithStyles(ListItem)>
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      key="settings"
      location={
        Object {
          "hash": "tEsT",
          "pathname": "/app",
          "search": "",
        }
      }
      to="/account/settings"
    >
      <WithStyles(ListItem)>
        <WithStyles(ListItemIcon)>
          <pure(AccountCircleIcon) />
        </WithStyles(ListItemIcon)>
        <WithStyles(ListItemText)
          disableTypography={true}
        >
          <FormattedMessage
            defaultMessage="Settings"
            id="app.routes.settings"
            values={Object {}}
          />
        </WithStyles(ListItemText)>
      </WithStyles(ListItem)>
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      key="logout"
      location={
        Object {
          "hash": "tEsT",
          "pathname": "/app",
          "search": "",
        }
      }
      to="/auth/logout"
    >
      <WithStyles(ListItem)>
        <WithStyles(ListItemIcon)>
          <pure(ExitToAppIcon) />
        </WithStyles(ListItemIcon)>
        <WithStyles(ListItemText)
          disableTypography={true}
        >
          <FormattedMessage
            defaultMessage="Logout"
            id="app.routes.logout"
            values={Object {}}
          />
        </WithStyles(ListItemText)>
      </WithStyles(ListItem)>
    </NavLink>
  </WithStyles(Drawer)>
</div>
`;

exports[`<Header /> when no user logged in... snapshot 1`] = `
<div>
  <WithStyles(AppBar)
    className=""
    position="absolute"
  >
    <WithStyles(Toolbar)
      disableGutters={true}
    >
      <WithStyles(IconButton)
        aria-label="Open drawer"
        color="inherit"
        onClick={[Function]}
      >
        <pure(MenuIcon) />
      </WithStyles(IconButton)>
      <nav>
        <div
          id="header-title"
        >
          <Link
            replace={false}
            to="/"
          >
            <WithStyles(Typography)
              variant="headline"
            >
              <FormattedMessage
                defaultMessage="dtodo1paco"
                id="app.title"
                values={Object {}}
              />
            </WithStyles(Typography)>
          </Link>
        </div>
        <div
          id="header-links"
        >
          <div>
            <NavLink
              activeClassName="active"
              aria-current="page"
              key="login"
              location={
                Object {
                  "hash": "tEsT",
                  "pathname": "/",
                  "search": "",
                }
              }
              to="/auth/login"
            >
              <WithStyles(ListItem)>
                <WithStyles(ListItemText)
                  disableTypography={true}
                >
                  <FormattedMessage
                    defaultMessage="Login"
                    id="app.routes.login"
                    values={Object {}}
                  />
                </WithStyles(ListItemText)>
              </WithStyles(ListItem)>
            </NavLink>
            <NavLink
              activeClassName="active"
              aria-current="page"
              key="help"
              location={
                Object {
                  "hash": "tEsT",
                  "pathname": "/",
                  "search": "",
                }
              }
              to="/help"
            >
              <WithStyles(ListItem)>
                <WithStyles(ListItemText)
                  disableTypography={true}
                >
                  <FormattedMessage
                    defaultMessage="Help"
                    id="app.routes.help"
                    values={Object {}}
                  />
                </WithStyles(ListItemText)>
              </WithStyles(ListItem)>
            </NavLink>
          </div>
          <WithStyles(Connect(LocaleToggle))
            inAppBar={true}
          />
        </div>
      </nav>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
  <WithStyles(Drawer)
    ModalProps={
      Object {
        "onBackdropClick": [Function],
      }
    }
    id="header-drawer-links"
    onClick={[Function]}
    open={false}
  >
    <div>
      <WithStyles(Typography)
        variant="headline"
      >
        <FormattedMessage
          defaultMessage="dtodo1paco"
          id="app.title"
          values={Object {}}
        />
      </WithStyles(Typography)>
      <WithStyles(IconButton)>
        <pure(ChevronLeftIcon) />
      </WithStyles(IconButton)>
    </div>
    <WithStyles(Divider) />
    <NavLink
      activeClassName="active"
      aria-current="page"
      key="login"
      location={
        Object {
          "hash": "tEsT",
          "pathname": "/",
          "search": "",
        }
      }
      to="/auth/login"
    >
      <WithStyles(ListItem)>
        <WithStyles(ListItemIcon)>
          <pure(AccountCircleIcon) />
        </WithStyles(ListItemIcon)>
        <WithStyles(ListItemText)
          disableTypography={true}
        >
          <FormattedMessage
            defaultMessage="Login"
            id="app.routes.login"
            values={Object {}}
          />
        </WithStyles(ListItemText)>
      </WithStyles(ListItem)>
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      key="help"
      location={
        Object {
          "hash": "tEsT",
          "pathname": "/",
          "search": "",
        }
      }
      to="/help"
    >
      <WithStyles(ListItem)>
        <WithStyles(ListItemIcon)>
          <pure(HelpIcon) />
        </WithStyles(ListItemIcon)>
        <WithStyles(ListItemText)
          disableTypography={true}
        >
          <FormattedMessage
            defaultMessage="Help"
            id="app.routes.help"
            values={Object {}}
          />
        </WithStyles(ListItemText)>
      </WithStyles(ListItem)>
    </NavLink>
  </WithStyles(Drawer)>
</div>
`;
